libs/cart/src/containers/cart/cart.component.ts

```diff
@@ -6,29 +6,27 @@ import { Store, select } from '@ngrx/store';
 import * as fromCart from '../../reducers/public_api';
 import { DaffCartLoad, DaffAddToCart } from '../../actions/cart.actions';
 import { DaffCart } from '../../models/cart';
-import {
-  selectCartLoading,
-  selectCartValue
-} from '../../selectors/public_api';
+import { getDaffCartSelectors } from '../../selectors/public_api';
+
 @Component({
   selector: '[cart-container]',
   template: '<ng-content></ng-content>',
   exportAs: 'CartContainer'
 })
-export class DaffCartContainer implements OnInit {
+export class DaffCartContainer<T extends DaffCart> implements OnInit {
 
   loading$: Observable<boolean>;
   cart$: Observable<DaffCart>;
 
   constructor(
-    private store: Store<fromCart.DaffCartReducersState>
+    private store: Store<fromCart.DaffCartReducersState<T>>
   ) { }
 
   ngOnInit() {
     this.store.dispatch(new DaffCartLoad());
 
-    this.loading$ = this.store.pipe(select(selectCartLoading));
-    this.cart$ = this.store.pipe(select(selectCartValue));
+    this.loading$ = this.store.pipe(select(getDaffCartSelectors<T>().selectCartLoading));
```

Comment:
Can the initial call be moved to a const either in the ngOnInit or constructor?

Reply:
done